### ISSUE TITLE
are there any documents can help me to understand how this software works?

### ISSUE THREAD

**Issue by rossihwang** [zeta0134/LuaGB#6]
It's an awesome project! I think it is possible to port to a embedded system! thank you

**Comment by dimiro1** [comment]
http://bgb.bircd.org/pandocs.htm
http://marc.rawer.de/Gameboy/Docs/GBCPUman.pdf

**Comment by zeta0134** [comment]
The two documents dimiro1 posted were my main references when emulating most of the system. I also used http://gbdev.gg8.se/wiki/articles/Gameboy_sound_hardware extensively for figuring out the finer details of the audio subsystem.

Documentation on the Gameboy's hardware is surprisingly hard to find! There's lots of general overviews and guides for programming, but the actual implementation details I worked out largely through trial and error.

If you're meaning to understand the LuaGB code itself, no there's not any documentation written. I try to write sane-looking source code though, so it should be decently easy to read and understand if you know Lua already. The magic starts in gameboy/init.lua, and each bit of the hardware has a module somewhere in the gameboy folder. In particular, the processor is located in gameboy/z80/init.lua, and should be a good starting point to understand how everything's put together.
